### PR TITLE
PLT-7542 Converting LDAP sync to the job server

### PR DESCRIPTION
**Pull request by coreyhulen**
EE PR https://github.com/mattermost/enterprise/pull/196

### CODE REVIEW

File: cmd/platform/ldap.go [hunk 1]
```diff
@@ -32,10 +33,11 @@ func ldapSyncCmdF(cmd *cobra.Command, args []string) error {
 	}
 
 	if ldapI := einterfaces.GetLdapInterface(); ldapI != nil {
-		if err := ldapI.Syncronize(); err != nil {
-			CommandPrintErrorln("ERROR: AD/LDAP Synchronization Failed")
+		job, err := ldapI.StartSynchronizeJob(true)
+		if err != nil || job.Status == model.JOB_STATUS_ERROR {
```
Comment (grundleborg) [inline cmd/platform/ldap.go]: Maybe this should check for "cancelled" as  well? In fact, I'm wondering if the jobs package could benefit from some utility functions like `didFinishSuccessfully()` and `didFinishUnsuccessfully()` and `hasNotFinishedYet()` etc. These would then ensure that whereever in the code we do checks like this, the different statuses are treated consistently.

File: cmd/platform/ldap.go [hunk 2]
```diff
@@ -32,10 +33,11 @@ func ldapSyncCmdF(cmd *cobra.Command, args []string) error {
...
+			CommandPrintErrorln("ERROR: AD/LDAP synchronization please check the server logs")
 		} else {
-			CommandPrettyPrintln("SUCCESS: AD/LDAP Synchronization Complete")
+			CommandPrettyPrintln("SUCCESS: AD/LDAP synchronization")
```
Comment (grundleborg) [inline cmd/platform/ldap.go]: any particular reason to remove the "complete" from the end of this?

File: jobs/jobs_watcher.go [hunk 3]
```diff
@@ -12,21 +12,23 @@ import (
 )
 
 const (
-	WATCHER_POLLING_INTERVAL = 15000
+	DEFAULT_POLLING_INTERVAL = 15000
```
Comment (grundleborg) [inline jobs/jobs_watcher.go]: Can you call this `DEFAULT_WATCHER_POLLING_INTERVAL` as I don't want it to get confused with the other polling intervals that exist in the `jobs` package?

File: model/job.go [hunk 4]
```diff
@@ -13,6 +13,7 @@ const (
 	JOB_TYPE_DATA_RETENTION                 = "data_retention"
 	JOB_TYPE_ELASTICSEARCH_POST_INDEXING    = "elasticsearch_post_indexing"
 	JOB_TYPE_ELASTICSEARCH_POST_AGGREGATION = "elasticsearch_post_aggregation"
+	JOB_TYPE_LDAP_SYNC                      = "lday_sync"
```
Comment (grundleborg) [inline model/job.go]: typo `lday_sync`

File: cmd/platform/ldap.go (hunk 1 above)
Comment (coreyhulen) [reply cmd/platform/ldap.go]: Yup, I can add cancelled.

File: cmd/platform/ldap.go (hunk 2 above)
Comment (coreyhulen) [reply cmd/platform/ldap.go]: hah, No good reason, I deleted it in an earlier PR and didn't add it back until later.  I can make it the original string.

File: jobs/jobs_watcher.go (hunk 3 above)
Comment (coreyhulen) [reply jobs/jobs_watcher.go]: Yup

File: model/job.go (hunk 4 above)
Comment (coreyhulen) [reply model/job.go]: thanks, I will fix.

**Comment by coreyhulen** [comment]
@grundleborg all the changes requested have been made.

**Comment by crspeller** [comment]
@coreyhulen Need a rebase, then feel free to merge.

**Pull request merged**